src/NumericalAlgorithms/SphericalHarmonics/Strahlkorper.cpp

```diff
@@ -34,17 +34,27 @@ Strahlkorper<Frame>::Strahlkorper(const size_t l_max, const size_t m_max,
 template <typename Frame>
 Strahlkorper<Frame>::Strahlkorper(
     const size_t l_max, const size_t m_max,
-    const DataVector& radius_at_collocation_points,
-    std::array<double, 3> center)
+    const DataVector& collocation_radii_or_spectral_coefs,
+    std::array<double, 3> center, const StrahlkorperConstructorData data_type)
```

Comment:
To avoid passing this enum to every construction of a Strahlkorper you could distinguish the new constructor by taking a `ModalVector spectral_coefs`. Then the correct constructor gets selected automatically, and `ModalVector` is the right data structure for the coefficients anyway.

Reply:
I think choosing this method will lead to a lot of unrelated changes to this PR that are not necessary. Namely that if we add this constructor, we should also change the other two constructors that take spectral coefs as `DataVector`s and change them to `ModalVector`s, and consequently everywhere those constructors are used. However, this will require a lot of copying between `DataVector` and `ModalVector` both externally and internally. Then there is the question of if we change the internal storage of the coefs to `ModalVector` instead (since that's actually the right data structure). This would mean several `Spherepack` functions would need to change, everywhere that uses those functions, the `coefficients()` function of `Strahlkorper` and everywhere that uses that.

Basically, if we go with this change, I'll insist we do it right and not part way which adds so many unrelated changes. Maybe we should make this an issue instead?

Reply:
I'm all for minimal changes, which is why I'm proposing this to avoid the added enum to _every_ construction of a Strahlkorper. Maybe there's a middle ground that avoids having to refactor much of Strahlkorper. I'd be fine with copying the ModalVector to the internal DataVector in the new constructor for now, with a comment that the copy can be avoided when we switch the internal storage to ModalVector. The new constructor is only used when loading the Strahlkorper from disk at the moment, so the copy doesn't hurt.

Reply:
Then we open an issue to do the refactor to ModalVector and do it as a follow-up.

Reply:
Ok yeah I'm fine with copying the `ModalVector` into the internal `DataVector`. We'll probably need to make both the new `ModalVector` version and old `DataVector` version of that constructor `explicit` to avoid any unwanted conversion.